nengo/connection.py

```diff
@@ -117,25 +117,70 @@ def validate(self, conn, distorarray):
 class ConnectionFunctionParam(FunctionParam):
     """Connection-specific validation for functions."""
 
+    def __set__(self, conn, function):
+        if function is None:
+            function_info = self.Info(function=None, size=None)
+        elif is_array_like(function):
+            array = np.array(function, copy=False, dtype=np.float64)
+            self.validate_array(conn, array)
+            function_info = self.Info(function=array, size=array.shape[1])
+        elif callable(function):
+            function_info = self.Info(function=function,
+                                      size=self.determine_size(conn, function))
+            self.validate_callable(conn, function_info)
+        else:
+            raise ValidationError("Invalid connection function type %r "
+                                  "(must be callable or array-like)"
+                                  % function.__class__.__name__,
```

Comment:
I think there are only a couple rare cases in Python 2 where this attribute won't exist. I'm not really up to snuff on Python 3 so I'm afraid I don't know what the different cases are, or what the best suggestion would be. `type` does sound more reliable, but again I don't know.